python/hsfs/feature_group.py

```diff
@@ -2189,6 +2194,47 @@ def __init__(
         self._feature_writers: Optional[Dict[str, callable]] = None
         self._writer: Optional[callable] = None
         self._kafka_headers: Optional[Dict[str, bytes]] = None
+        # On-Demand Transformation Functions
+        self._transformation_functions: List[TransformationFunction] = []
+
+        if transformation_functions:
```

Comment:
need to consider transformation in `feature_group.save` as well, when feature are defined in `create_feature_group`.

Reply:
Updated to handle cases where create_feature_group is defined with features and feature_group.save if called.

To handle this usecase I moved the function to updating the schema based on transformation functions from `python/spark engine` to `feature_group_engine`. In `feature_group_engine` the `_update_feature_group_schema_on_demand_transformations` function updates the schema to add transformed features and drop any feature if specified in the transformation function.